hw/ip/otbn/dv/uvm/gen-binaries.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+'''A helper script to generate a default set of binaries for OTBN testing
+
+This is intended for use with dvsim, which should call this script as
+part of the test build phase.
+
+'''
+
+import argparse
+import os
+import subprocess
+import sys
+import tempfile
+
+
+def read_positive(val):
+    ival = -1
+    try:
+        ival = int(val, 0)
+    except ValueError:
+        pass
+
+    if ival <= 0:
+        raise argparse.ArgumentTypeError('{!r} is not a positive integer.'
+                                         .format(val))
+    return ival
+
+
+def read_jobs(val):
+    if val == 'unlimited':
+        return 'unlimited'
+    if val is None:
+        return None
+
+    return read_positive(val)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--count', type=read_positive, default=10,
+                        help='Number of binaries to generate')
+    parser.add_argument('--seed', type=read_positive, default=0)
+    parser.add_argument('--jobs', '-j', type=read_jobs, nargs='?',
+                        const='unlimited', help='Number of parallel jobs.')
+    parser.add_argument('destdir',
+                        help='Destination directory')
+
+    args = parser.parse_args()
+    if args.count < 0:
+        print('ERROR: Any argument to --count must be positive.',
+              file=sys.stderr)
+        return 1
+
+    otbn_dir = os.path.normpath(os.path.join(os.path.dirname(__file__),
+                                             '../..'))
+
+    rig_count = args.count - 1
+    util_dir = os.path.join(otbn_dir, 'util')
+    smoke_src_dir = os.path.join(otbn_dir, 'dv/smoke')
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        with open(os.path.join(tmpdir, 'Makefile'), 'w') as make_handle:
+            write_makefile(make_handle, rig_count, args.seed,
```

Comment:
You may want the pass the same simulation run seed to this script to ensure reproducibility.

Reply:
Hmm, maybe. The problem is that if we're treating "build some binaries" as part of the build step (rather than the run step), we don't yet have a seed floating around. This question probably better belongs in the "how do we integrate `build-me-some-otbn-binaries.sh` with dvsim?" discussion, maybe in PR #4189.